src/OrbitQt/servicedeploymanager.cpp

```diff
@@ -280,12 +283,18 @@ outcome::result<void> ServiceDeployManager::CopyOrbitServicePackage() {
 
 ErrorMessageOr<void> ServiceDeployManager::CopyFileToLocal(std::string source,
                                                            std::string destination) {
-  ErrorMessageOr<void> result = outcome::success();
+  orbit_base::Promise<ErrorMessageOr<void>> promise;
+  auto future = promise.GetFuture();
+
   DeferToBackgroundThreadAndWait(
-      this, [&, source = std::move(source), destination = std::move(destination)]() {
-        result = CopyFileToLocalImpl(source, destination);
+      this, [this, source = std::move(source), destination = std::move(destination),
+             promise = std::move(promise)]() mutable {
+        promise.SetResult(CopyFileToLocalImpl(source, destination));
       });
-  return result;
+
+  if (future.IsFinished()) return future.Get();
```

Comment:
personal prefference, you can ignore it:  if (!future ... ) return ErrorMessageOr(...);

Reply:
This is actually also my personal preference. Don't know why it wrote it differently.

Reply:
Fixed. Thanks.